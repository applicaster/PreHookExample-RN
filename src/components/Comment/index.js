import React, { Component } from 'react';
import PropTypes from 'prop-types';
import reactStringReplace from 'react-string-replace';
import {
  StyleSheet,
  Text,
} from 'react-native';

const styles = StyleSheet.create({
  comment: {
    fontSize: 14,
    paddingHorizontal: 12,
    paddingTop: 6,
    paddingBottom: 12,
    lineHeight: 18,
  },
});

class Comment extends Component {
  highlightHashtagsAndUsers(caption) {
    const highlightColor = { color: this.context.mainColor };
    const regex = /([@|#][\w](?:(?:[\w]|(?:\.(?!\.))){0,28}(?:[\w]))?)/g;
    const matchCallback = (match, i) => (<Text key={i} style={highlightColor}>{match}</Text>);
    return reactStringReplace(caption, regex, matchCallback);
  }

  render() {
    const { caption } = this.props;
    const captionColor = { color: this.context.textColor };
    return <Text style={[styles.comment, captionColor]}>{this.highlightHashtagsAndUsers(caption)}</Text>;
  }
}

Comment.propTypes = {
  caption: PropTypes.string,
};

Comment.contextTypes = {
  mainColor: PropTypes.string,
  textColor: PropTypes.string,
};

export default Comment;
