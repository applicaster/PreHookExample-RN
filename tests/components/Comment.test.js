import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Comment from '../../src/components/Comment';

describe('Comment', () => {
  test('component renders correctly with plain text', () => {
    const html = renderer.create(
      <Comment caption="Hola Carlos Pinto, great picture." />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly with hashtags and usertags', () => {
    const html = renderer.create(
      <Comment caption="Hola @cpinto7, #great #picture" />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
