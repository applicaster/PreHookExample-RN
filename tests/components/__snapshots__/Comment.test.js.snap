// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component renders correctly with hashtags and usertags 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontSize": 14,
        "lineHeight": 18,
        "paddingBottom": 12,
        "paddingHorizontal": 12,
        "paddingTop": 6,
      },
      Object {
        "color": undefined,
      },
    ]
  }
>
  Hola 
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": undefined,
      }
    }
  >
    @cpinto7
  </Text>
  , 
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": undefined,
      }
    }
  >
    #great
  </Text>
   
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": undefined,
      }
    }
  >
    #picture
  </Text>
  
</Text>
`;

exports[`Comment component renders correctly with plain text 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontSize": 14,
        "lineHeight": 18,
        "paddingBottom": 12,
        "paddingHorizontal": 12,
        "paddingTop": 6,
      },
      Object {
        "color": undefined,
      },
    ]
  }
>
  Hola Carlos Pinto, great picture.
</Text>
`;
